transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueueDomainSocketChannelConfig.java

```diff
@@ -47,6 +49,9 @@
         if (option == DOMAIN_SOCKET_READ_MODE) {
             return (T) getReadMode();
         }
+        if (option == ALLOW_HALF_CLOSURE) {
```

Comment:
I think you also need to add it to the `getOptions` method. See : https://github.com/netty/netty/blob/1c66815c8aadb53f373e8070279a510a2eda89a7/transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollSocketChannelConfig.java#L60